Sprint_6

AntonArattsevYandex opened the pull request.

AntonArattsevYandex committed:
Выполнение задания по тестированию

pomegranate282922 commented on src/test/java/AnimalTest.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Test(expected = Exception.class)
+    public void getFood_ShouldThrowException_ForUnknownType() throws Exception {
```
camelCase 
Нужно исправить

pomegranate282922 commented on src/test/java/LionParameterizedTest.java:
```diff
@@ -0,0 +1,32 @@
+    @Parameterized.Parameters(name = "Для пола {0} наличие гривы должно быть {1}")
+    public static Collection<Object[]> testCases() {
+        return Arrays.asList(new Object[][]{
+                {"Самец", true},
```
Отлично применена параметризация

pomegranate282922 reviewed (commented):
Привет! Все отлично, только не сделана изоляция класса Lion от Feline.

AntonArattsevYandex committed:
Исправлены замечания:
1. Приведены в camelCase названия методов тестов
2. Реализована инъекция зависимостей для класса Lion через конструктор

Старый код сохранён для обратной совместимости.
Все тесты проходят, покрытие 100%.

pomegranate282922 commented on src/main/java/com/example/Lion.java:
```diff
@@ -16,7 +18,10 @@ public Lion(String sex) throws Exception {
 
-    Feline feline = new Feline();
+    // Старый конструктор (оставлен для совместимости)
+    public Lion(String sex) throws Exception {
```
Нет, таким образом класс не полностью изолирован. Нужно убрать все связи.
Нужно исправить

pomegranate282922 reviewed (commented):
Все ок, но Lion оказался не полностью изолированным.

AntonArattsevYandex committed:
fix: Полная изоляция Lion от Feline через DI

Что сделано:
1. Удалён старый конструктор Lion(String sex)
2. Lion теперь зависит только от интерфейса Predator
3. Feline реализует Predator (как и раньше)
4. Тесты используют реальный Feline (без моков)

Изменения минимальны:
- Только правки для DI в Lion.java
- В тестах только добавлен параметр Feline в конструктор
- Вся старая логика сохранена

Все тесты проходят, покрытие 100%.

pomegranate282922 reviewed (commented):
Ок, зачет.
В тестах на lion лучше тоже применить Моки.
